tests/test_logic.py

```diff
@@ -394,3 +393,32 @@ def test_finite(op, data, dtype):
     dpnp_res = getattr(dpnp, op)(x, out=dp_out)
     assert dp_out is dpnp_res
     assert_equal(dpnp_res, np_res)
+
+
+@pytest.mark.parametrize("func", ["isneginf", "isposinf"])
+@pytest.mark.parametrize(
+    "data",
+    [
+        [dpnp.inf, -1, 0, 1, dpnp.nan, -dpnp.inf],
+        [[dpnp.inf, dpnp.nan], [dpnp.nan, 0], [1, -dpnp.inf]],
+    ],
+    ids=[
+        "1D array",
+        "2D array",
+    ],
+)
+@pytest.mark.parametrize("dtype", get_float_complex_dtypes())
+def test_infinity_sign(func, data, dtype):
+    x = dpnp.asarray(data, dtype=dtype)
+    if dpnp.issubdtype(dtype, dpnp.complexfloating):
+        with pytest.raises(TypeError):
```

Comment:
I would propose to split into two tests. In the 1st one we will need to check that numpy and dpnp both raises the same exception.